weave/flow/scorers/base_scorer.py

```diff
@@ -0,0 +1,109 @@
+from numbers import Number
+from typing import Any, Callable, Optional, Sequence, Tuple, Union
+
+import numpy as np
+from pydantic import BaseModel, Field
+
+import weave
+from weave.flow.obj import Object
+from weave.trace.isinstance import weave_isinstance
+from weave.trace.op import Op, as_op, is_op
+
+
+class Scorer(Object):
```

Comment:
is it helpful to inherit from Object?  Would you be against making this e.g. a normal class or dataclass?

Reply:
Inheritance from Object is how its defined today in master. We have validators in `LLMScorer` that rely on this being a pydantic basemodel though so BaseModel would have to be the alternative I guess.

Reply:
Is it helpful to inherit from BaseModel?  Are you using any pydantic-specific features here?